pkg/framework/machines.go

```diff
@@ -118,13 +120,16 @@ func DeleteMachine(client runtimeclient.Client, machine *mapiv1beta1.Machine) er
 func WaitForMachineDelete(c client.Client, machines ...*mapiv1beta1.Machine) {
```

Comment:
If this is now iterating, perhaps we should call it `WaitForMachinesDeleted`?

Reply:
same for `DeleteMachine`